samples/sample_common/src/sample_utils.cpp

```diff
@@ -1270,6 +1309,83 @@ mfxStatus CSmplYUVWriter::WriteNextFrameI420(mfxFrameSurface1 *pSurface)
     return MFX_ERR_NONE;
 }
 
+void QPFile::Reader::ResetState()
+{
+    ResetState(READER_ERR_NOT_INITIALIZED);
+}
+
+void QPFile::Reader::ResetState(ReaderStatus set_sts)
+{
+    m_CurFrameNum = std::numeric_limits<mfxU32>::max();
+    m_nFrames     = std::numeric_limits<mfxU32>::max();
+    m_ReaderSts   = set_sts;
+    m_FrameVals.clear();
+}
+
+mfxStatus QPFile::Reader::Read(const msdk_string& strFileName, mfxU32 codecid)
+{
+    m_ReaderSts   = READER_ERR_NONE;
+    m_CurFrameNum = 0;
+
+    if (codecid != MFX_CODEC_AVC && codecid != MFX_CODEC_HEVC)
+    {
+        ResetState(READER_ERR_CODEC_UNSUPPORTED);
+        return MFX_ERR_NOT_INITIALIZED;
+    }
+    
+    std::ifstream ifs(strFileName, msdk_fstream::in);
+    if (!ifs.is_open())
+    {
+        ResetState(READER_ERR_FILE_NOT_OPEN);
+        return MFX_ERR_NOT_INITIALIZED;
+    }
+
+    try
+    {
+        FrameInfo   frameInfo{};
+        std::string line;
+        std::getline(ifs, line);
+        m_nFrames = std::stoi(line); // number of frames at first line
+
+        m_FrameVals.reserve(m_nFrames);
+
+        while (QPFile::get_line(ifs, line))
+        {
+            frameInfo.displayOrder = QPFile::ReadDisplayOrder(line);
+            frameInfo.QP = QPFile::ReadQP(line);
+            frameInfo.frameType = QPFile::ReadFrameType(line);
+            if ( frameInfo.displayOrder > m_nFrames ||
+                 frameInfo.QP > 51 ||
+                 frameInfo.frameType == MFX_FRAMETYPE_UNKNOWN )
+            {
+                ResetState(READER_ERR_INCORRECT_FILE);
+                return MFX_ERR_NOT_INITIALIZED;
+            }
+            m_FrameVals.push_back(frameInfo);
+        }
+        if (m_FrameVals.size() < m_nFrames)
+        {
+            ResetState(READER_ERR_INCORRECT_FILE);
+            return MFX_ERR_NOT_INITIALIZED;
+        }
+    }
+    catch (std::exception& /*e*/)
```

Comment:
are you trying to catch some specific exception?

Reply:
there may be many possible exceptions, i think it is better to leave the common exception

Reply:
Every sample has it's own try/catch block on top level. I don't see any point in local exception handling. Putting try/catch block in every function where exception may occur is not a correct thing as for me. I'd say that you need to add exception handling only if you need to process all errors locally at current level. It is not definitely this case